### ISSUE TITLE
Suggestion: Reference spec document links in the meta-schema

### ISSUE THREAD

**Issue by LouizFC** [json-schema-org/json-schema-spec#1119]
Sorry if this is a duplicate.

I think that linking the specific spec section url in each meta vocabularies keyword would be very helpful.

I am in the process of implementing json schema in a proprietary language and I think it would be a friendly way to document each keyword semantics without the need to manually search for it in the spec documents

**Comment by Relequestual** [comment]
Hi an welcome!
Interesting idea, but it would cause a more bloated meta-schema, and that thing gets downloaded a LOT. We're talking GBs a day.

It could still be useful as an alternate schema, but I'm not sure we want to support another meta-schema just for linking to the spec. Plus, many keywords would a handful of links, as there's not always one section required to understand.

Further, we have our learning resources, which include all the keywords in a more accessible format 😅: https://json-schema.org/understanding-json-schema/

**Comment by LouizFC** [comment]
I see. You rise very good points. I think one alternate schema that maps keywords to their documentation would be nice, but also not as necessary to justify rising more problems.

Feel free to close the issue, thank you very much

**Comment by Relequestual** [comment]
You're welcome. We still appreciate you raising the issue! Say it how you see it.

We could do better at directing people to our learning resources.
Any suggestions on how we could do that? We have a few ideas, but more couldn't hurt!

**Comment by LouizFC** [comment]
Sorry for the delayed response.

Some tools (vscode for example) can lint and validate json files based on json schemas. Most schemas have detailed descriptions about their properties ([tsconfig.json schema as an example](http://json.schemastore.org/tsconfig)).

With that in mind, an alternative schema (potentially auto filled by templating tools?) that has either spec links or the spec itself embedded within could be useful for people who are not familiar with the spec. 

```json
...
  "type": {
      "description": "... embedded spec description ...",
      "$comment": "See also: https://json-schema.org/draft/2020-12/json-schema-core.html#rfc.section.4.2.1",
      "anyOf": [
        { "$ref": "#/$defs/simpleTypes" },
        {
          "type": "array",
          "items": { "$ref": "#/$defs/simpleTypes" },
          "minItems": 1,
          "uniqueItems": true
        }
      ]
    },
...
```
You mentioned that some keywords would require the schema author to read multiple sections of the spec. I think that this could be solved by writing something similar to "see also" description inside `$comment`

You raised a bandwidth concern which I can't think of an objective solution now, but maybe having a disclaimer discouraging the use of the "full doc" schema in some way could be a start

**Comment by karenetheridge** [comment]
I would not be opposed to adding a single URL per vocabulary document, linking to the high level section where that vocabulary is described.

These are the links for draft2020-12:

applicator - https://datatracker.ietf.org/doc/html/draft-bhutton-json-schema-00#section-10
content- https://datatracker.ietf.org/doc/html/draft-bhutton-json-schema-validation-00#section-8
core - https://datatracker.ietf.org/doc/html/draft-bhutton-json-schema-00#section-8
format-annotation - https://datatracker.ietf.org/doc/html/draft-bhutton-json-schema-validation-00#section-7
format-assertion - https://datatracker.ietf.org/doc/html/draft-bhutton-json-schema-validation-00#section-7
meta-data - https://datatracker.ietf.org/doc/html/draft-bhutton-json-schema-validation-00#section-9
unevaluated - https://datatracker.ietf.org/doc/html/draft-bhutton-json-schema-00#section-11

**Comment by LouizFC** [comment]
Thanks for the feedback.

I think the point I am trying to express is that json schema keywords introduce a lot of cognitive load, and making it "self-documented" would decrease it considerably (that is just my opinion of course).

Introducing cognitive load is inevitable, but I think that some "summarized" or even "embedded as is" description in the schema itself could help lessen it. Links are useful for sure, but most programming languages (not saying that json schema is a programming language, I understand its scope as a specification) the documentation is usually in the same place as the code itself.

In summary:

I think the best approach without introducing any new schema is just linking the sections as @karenetheridge suggested.

A full fledged "schema.doc.json" with more in-depth descriptions would be useful, but I understand that doing this would introduce another document that needs to be maintained, so I don't think it is worth the effort unless generating such document can be automated.
